test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -223,6 +222,46 @@ object TestAspect extends TimeoutVariants {
    */
   val flaky: TestAspectPoly = eventually
 
+  /**
+   * An aspect that runs a test on a forked fiber. This is handy to use with aspects that repeat tests such as
+   * `nonFlaky` and `eventually` to avoid ever increasing `fiberTime`
+   */
+  val fork: TestAspectPoly =
+    new TestAspect.PerTest[Nothing, Any, Nothing, Any, Nothing, Any] {
+
+      def perTest[R >: Nothing <: Any, E >: Nothing <: Any, S >: Nothing <: Any](
+        test: ZIO[R, E, Either[TestFailure[Nothing], TestSuccess[S]]]
+      ): ZIO[R, E, Either[TestFailure[Nothing], TestSuccess[S]]] =
+        test.fork.flatMap(_.await).flatMap(ZIO.done)
+    }
+
+  /**
+   * An aspect that runs each test on a separate fork and resets the TestClock to 0.
```

Comment:
Should we generalize to `resetTestEnvironment` and provide fresh copies of all the standard ZIO environment types?

Reply:
I just merged #2043 which will provide some additional tools for us to do this in a clean way.